lib/change_stream.js

```diff
@@ -158,7 +160,7 @@ class ChangeStream extends EventEmitter {
     if (this.cursor) {
       return this.cursor.isClosed();
     }
-    return true;
+    return this.closed;
```

Comment:
I think we should _only_ return `this.close`, instead of deferring to the underlying cursor.

Reply:
I tried to do this, but the test `Should invalidate change stream on database drop using imperative callback form` starts failing. The underlying cursor closes due to the `invalidate` event, but `ChangeStream` doesn't have any way of knowing that except calling `this.cursor.isClosed()`. I suspect there's a way around that, but the current approach seems to work well enough.

Reply:
Hmm, okay yeah that makes sense. If the cursor receives a `getMore` response with `cursorId: 0` then it will close the cursor but we won't find out about that until we call this. We'd only be able to track the `close` event, but that would force the cursor into streaming mode.